# STAT 207 Lab Assignment 10 - Group Part - [9 Points]



## <u>Case Study</u>: Predicting Airbnb Superhost Status

Suppose that you have a spare room in your house (in Seattle) and you are considering advertising this room to guests on Airbnb. Airbnb superhosts are considered to be experienced hosts who provide a shining example for other hosts, and extraordinary experiences for their guests. Once a host reaches Superhost status, a superhost badge will automatically appear on their listing and profile to help guests identify them. You would like to assess your chances of being named a superhost with your property.

The following dataset is a sample (assume random) of available Airbnb listings in Seattle, WA. These listings were collected in January 2016, and filtered to just contain listings from the five most popular Seattlbe neighborhoods (for Airbnb listings) and just contain listings that are either in a house or apartment property.

This dataset contains the following variables.

**Listing Information**
The dataset contains the following information about the Airbnb *listing*:
* <u>price</u>: price of the listing (per night)(in US dollars)
* <u>review_scores_rating</u>: the average rating of the listing [0,100] (100 is the best)
* <u>number_of_reviews</u>: the number of reviews for the listing
* <u>security_deposit</u>: the security deposit required for the listing (in US dollars)
* <u>cleaning_fee</u>: the cleaning fee required for the listing (in US dollars)
* <u>neighborhood</u>: the neighborhood of Seattle the listing is located in
* <u>property_type</u>: is the listing in a 'House' or 'Apartment'
* <u>room_type</u>: is the listing a 'Entire home/apt', 'Private room', or 'Shared room'
* <u>accommodates</u>: how many guests will the listing accommodate
* <u>bathrooms</u>:how many bathrooms does the listing have
* <u>beds</u>: how many beds does the listing have

**Host Information**
The dataset also contains the following information about the *host* of the given Airbnb listing:
* <u>host_is_superhost</u>: is the host a "superhost": t=True, f=False
* <u>host_has_profile_pic</u>: does the host have a profile pic in their bio: t=True, f=False
* <u>host_response_time</u>: how fast will the host respond to requests (on average)

## Research Goals

The primary research goal that we will pursue in this lab will be to build a predictive model that effectively predicts the `probability a Seattle Airbnb host is a **superhost**` for *new Seattle Airbnb listings/hosts* given some combination of the `remaining explanatory variables`.



## Points

<p>&nbsp;</p>
<table style="border: none;border-collapse: collapse;width:102pt;">
    <tbody>
        <tr>
            <td style="color:black;font-size:15px;font-weight:700;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:general;vertical-align:bottom;border:.5pt solid black;height:14.25pt;width:51pt;">Problem</td>
            <td style="color:black;font-size:15px;font-weight:700;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:general;vertical-align:bottom;border:.5pt solid black;border-left:none;width:51pt;">Points</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;height:14.25pt;border-top:none;">1.1</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;border-top:none;border-left:none;">0.25</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;height:14.25pt;border-top:none;">1.2</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;border-top:none;border-left:none;">0.75</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;height:14.25pt;border-top:none;">1.3</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;border-top:none;border-left:none;">0.5</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;height:14.25pt;border-top:none;">2.1</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;border-top:none;border-left:none;">1</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;height:14.25pt;border-top:none;">2.2</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;border-top:none;border-left:none;">0.75</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;height:14.25pt;border-top:none;">2.3</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;border-top:none;border-left:none;">0.25</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;height:14.25pt;border-top:none;">2.4</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;border-top:none;border-left:none;">1</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;height:14.25pt;border-top:none;">3</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;border-top:none;border-left:none;">3</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;height:14.25pt;border-top:none;">4</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;border-top:none;border-left:none;">1.5</td>
        </tr>
    </tbody>
</table>

In [26]:
import pandas as pd
import numpy as np

## 1. Basic Data Examination and Manipulation

### 1.1. Basics
1. Read this csv file into a dataframe.

2. Display your dataframe.

3. Show how many rows your dataframe has.



In [27]:
airbnb = pd.read_csv('seattle_airbnb_listings_cleaned.csv')

In [28]:
airbnb_head = airbnb.head()
num_rows = len(airbnb)

airbnb_head, num_rows

(   price  review_scores_rating  number_of_reviews  security_deposit  \
 0    300                   100                 24               500   
 1    149                    96                 11               300   
 2     95                    95                 79               150   
 3    105                   100                 13               500   
 4    140                    99                 30               250   
 
    cleaning_fee neighborhood property_type        room_type  accommodates  \
 0            95  Wallingford         House  Entire home/apt             5   
 1           105  Wallingford     Apartment  Entire home/apt             6   
 2            40  Wallingford     Apartment  Entire home/apt             3   
 3            50  Wallingford         House     Private room             2   
 4            65  Wallingford         House  Entire home/apt             2   
 
    bathrooms  beds  host_response_time  host_response_rate host_is_superhost  \
 0        1.5  

### 1.2. 0/1 Response Variable

Create a new 0/1 variable in your dataset in which `superhosts=1` and `nonsuperhosts=0`. 

In [29]:
airbnb['host_is_superhost'] = airbnb['host_is_superhost'].replace({'t':1, 'f':0})
airbnb.head()

price  review_scores_rating  number_of_reviews  security_deposit  \
0    300                   100                 24               500   
1    149                    96                 11               300   
2     95                    95                 79               150   
3    105                   100                 13               500   
4    140                    99                 30               250   

   cleaning_fee neighborhood property_type        room_type  accommodates  \
0            95  Wallingford         House  Entire home/apt             5   
1           105  Wallingford     Apartment  Entire home/apt             6   
2            40  Wallingford     Apartment  Entire home/apt             3   
3            50  Wallingford         House     Private room             2   
4            65  Wallingford         House  Entire home/apt             2   

   bathrooms  beds  host_response_time  host_response_rate  host_is_superhost  \
0        1.5     3  within a few hours                 1.0                  1   
1        1.0     3      within an hour                 1.0                  0   
2        1.0     2      within an hour                 1.0                  0   
3        2.0     1  within a few hours                 1.0                  1   
4        1.0     1      within an hour                 1.0                  1   

  host_has_profile_pic host_identity_verified  
0                    t                      t  
1                    t                      t  
2                    t                      t  
3                    t                      t  
4                    t                      t

### 1.3. Train-Test-Split

Create a training dataset and a test dataset from your dataframe using a random state of `101` and a test dataset size that is a random sample of about 20% of the observations from the original dataset.

In [30]:
from sklearn.model_selection import train_test_split

# Define X and y

X = airbnb[['review_scores_rating', 'number_of_reviews', 'neighborhood', 'room_type', 'property_type']]
y = airbnb['host_is_superhost']

# One-Hot Encode Categorical Variables

X_encoded = pd.get_dummies(X, drop_first=True)

# Split the Data into Training and Testing Sets

X_train, X_test, y_train, y_test = train_test_split(X_encoded, y, test_size=0.20, random_state=101)
# Display the first few rows of the training data
X_train.head(), y_train.head()

(     review_scores_rating  number_of_reviews  neighborhood_Broadway  \
 231                    85                255                   True   
 259                    99                 21                   True   
 165                    95                  4                  False   
 89                    100                  4                  False   
 295                    96                 67                  False   
 
      neighborhood_Fremont  neighborhood_Minor  neighborhood_Wallingford  \
 231                 False               False                     False   
 259                 False               False                     False   
 165                 False               False                     False   
 89                  False                True                     False   
 295                  True               False                     False   
 
      room_type_Private room  room_type_Shared room  property_type_House  
 231                   False     

## 2. Full Model and Test Log-Likelihood Function

In the interest of time, let's only consider using 5 potential explanatory variables in our logistic regression model, which will predict the `probability that a host is a superhost`:

* `review_scores_rating`
* `number_of_reviews`
* `neighborhood`
* `room_type`
* `property_type`

### 2.1. Fit the Model

Fit this "full" logistic regression model and show the summary output table. Use this summary output table to find the log-likelihood function value of this model and the training dataset.

In [31]:
import statsmodels.api as sm
# Converting boolean values to integers
X_train_int = X_train.astype(int)

# Adding a constant to the features in the training set
X_train_const = sm.add_constant(X_train_int)

# Fitting the logistic regression model
log_reg_full = sm.Logit(y_train, X_train_const).fit()

# Showing the summary output
log_reg_full.summary()


Optimization terminated successfully.
         Current function value: 0.484682
         Iterations 8


<class 'statsmodels.iolib.summary.Summary'>
"""
                           Logit Regression Results                           
==============================================================================
Dep. Variable:      host_is_superhost   No. Observations:                  271
Model:                          Logit   Df Residuals:                      261
Method:                           MLE   Df Model:                            9
Date:                Tue, 31 Oct 2023   Pseudo R-squ.:                  0.1879
Time:                        12:11:36   Log-Likelihood:                -131.35
converged:                       True   LL-Null:                       -161.74
Covariance Type:            nonrobust   LLR p-value:                 9.501e-10
============================================================================================
                               coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------
const                      -25.5990      5.833     -4.388      0.000     -37.032     -14.166
review_scores_rating         0.2586      0.060      4.340      0.000       0.142       0.375
number_of_reviews            0.0148      0.005      3.260      0.001       0.006       0.024
neighborhood_Broadway       -1.1456      0.451     -2.542      0.011      -2.029      -0.262
neighborhood_Fremont        -0.4944      0.534     -0.926      0.354      -1.540       0.552
neighborhood_Minor           0.0453      0.553      0.082      0.935      -1.039       1.130
neighborhood_Wallingford    -0.5505      0.544     -1.011      0.312      -1.618       0.516
room_type_Private room       0.2990      0.447      0.669      0.504      -0.577       1.175
room_type_Shared room       -1.2377      1.117     -1.108      0.268      -3.427       0.951
property_type_House         -0.3487      0.403     -0.865      0.387      -1.139       0.441
============================================================================================
"""

### Interpretation of the Results:

1. **Log-Likelihood**: The log-likelihood of the model is -131.35, which gives us an idea of how well the model explains the data. The closer to 0, the better.
2. **Pseudo R-squared**: The value is 0.1879, indicating that around 18.79% of the variability in the response variable is explained by the model.
3. **Coefficients**: 
   - **const**: The intercept term is -25.5990, which is the log-odds of being a superhost when all other variables are 0.
   - **review_scores_rating**: For each additional point in review score rating, the log-odds of being a superhost increase by 0.2586.
   - **number_of_reviews**: For each additional review, the log-odds of being a superhost increase by 0.0148.
   - Other variables also have their respective coefficients, indicating their relationship with the response variable.
4. **P-values**: 
   - Variables such as `review_scores_rating`, `number_of_reviews`, and `neighborhood_Broadway` have p-values less than 0.05, suggesting that they are statistically significant predictors of being a superhost.

### 2.2. Training Data Predictive Probabilities

Calculate the predictive probabilities for all observations in the **training** dataset with model. **Call the pandas series that contains each of these predictive probabilities `phat_train`**.

*Note: In addition to using the **.predict()** function to predict the **predictive probability** of a single observation with a logistic regression model, we can use **.predict(YOUR_DATAFRAME)** to predict the predictive probabilities for all observations in a given dataframe with a logistic regression model.*

In [32]:
# Calculating the predictive probabilities for the training data
phat_train = log_reg_full.predict(X_train_const)

# Showing the first few values of phat_train
phat_train.head()


231    0.269478
259    0.303714
165    0.274971
89     0.505077
295    0.348822
dtype: float64

### 2.3. Training Log-Likelihood Function Value

Assuming that your training dataframe was called `df_train`, the code below should calculate the the log-likelihood function value for your training dataset and this model. (If not, modify the code below to contain the actual name of your training dataframe).

This **log_loss()** should take take the following values:
* the actual 0/1 response variable values
* the predictive probabilities of a given model

Verify that this output here is the same as what you got in 2.1.

In [33]:
from sklearn.metrics import log_loss

# Calculating the logistic loss (negative log-likelihood)
logistic_loss = log_loss(y_train, phat_train)

# Adjusting the sign to get the log-likelihood
log_likelihood = -logistic_loss * len(y_train)
log_likelihood


-131.3489333632751

The log-likelihood function value for the training dataset and the logistic regression model is approximately $−131.35$.

This value is very close to the log-likelihood value obtained in the summary output of the logistic regression model in step 2.1, confirming the consistency of the results.

### 2.4. Test Log-Likelihood Function Value

Use the code in 2.2 and 2.3 to now calculate the log-likelihood function value of the **test dataset** and this model.

In [36]:
X_test = X_test.astype(int)
# Adding a constant to the features in the test set
X_test_const = sm.add_constant(X_test)

# Calculating the predictive probabilities for the test dataset
phat_test = log_reg_full.predict(X_test_const)

# Showing the first few predictive probabilities
phat_test.head()


172    0.139903
38     0.496683
242    0.117410
64     0.546663
181    0.281868
dtype: float64

## 3. Backwards Elimination Algorithm

Ideally, we'd like to train (with our training dataset) and select the logistic regression model that will yield the HIGHEST **test log-likelihood function (LLF) value** (in our case it will be the LLF that is the least negative). Let's use a backwards elimination algorithm to try to help us find this model with the highest test LLF value.

*Note: Remember that the backwards elimination algorithm is just an algorithm "template". You can use this algorithm template to evaluate other types of models (like logistic regression) in attempt to find the model that has the highest metric value (like the highest test LLF value).*

In [37]:
# Showing the summary output of the full logistic regression model
log_reg_full.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                           Logit Regression Results                           
==============================================================================
Dep. Variable:      host_is_superhost   No. Observations:                  271
Model:                          Logit   Df Residuals:                      261
Method:                           MLE   Df Model:                            9
Date:                Tue, 31 Oct 2023   Pseudo R-squ.:                  0.1879
Time:                        12:14:57   Log-Likelihood:                -131.35
converged:                       True   LL-Null:                       -161.74
Covariance Type:            nonrobust   LLR p-value:                 9.501e-10
============================================================================================
                               coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------
const                      -25.5990      5.833     -4.388      0.000     -37.032     -14.166
review_scores_rating         0.2586      0.060      4.340      0.000       0.142       0.375
number_of_reviews            0.0148      0.005      3.260      0.001       0.006       0.024
neighborhood_Broadway       -1.1456      0.451     -2.542      0.011      -2.029      -0.262
neighborhood_Fremont        -0.4944      0.534     -0.926      0.354      -1.540       0.552
neighborhood_Minor           0.0453      0.553      0.082      0.935      -1.039       1.130
neighborhood_Wallingford    -0.5505      0.544     -1.011      0.312      -1.618       0.516
room_type_Private room       0.2990      0.447      0.669      0.504      -0.577       1.175
room_type_Shared room       -1.2377      1.117     -1.108      0.268      -3.427       0.951
property_type_House         -0.3487      0.403     -0.865      0.387      -1.139       0.441
============================================================================================
"""

In [38]:
# Dropping the variable with the highest p-value
X_train_reduced = X_train_const.drop(columns='neighborhood_Minor')
X_test_reduced = X_test_const.drop(columns='neighborhood_Minor')

# Refitting the logistic regression model without 'neighborhood_Minor'
log_reg_reduced = sm.Logit(y_train, X_train_reduced).fit()

# Showing the summary output of the reduced logistic regression model
log_reg_reduced.summary()


Optimization terminated successfully.
         Current function value: 0.484695
         Iterations 8


<class 'statsmodels.iolib.summary.Summary'>
"""
                           Logit Regression Results                           
==============================================================================
Dep. Variable:      host_is_superhost   No. Observations:                  271
Model:                          Logit   Df Residuals:                      262
Method:                           MLE   Df Model:                            8
Date:                Tue, 31 Oct 2023   Pseudo R-squ.:                  0.1879
Time:                        12:15:05   Log-Likelihood:                -131.35
converged:                       True   LL-Null:                       -161.74
Covariance Type:            nonrobust   LLR p-value:                 3.293e-10
============================================================================================
                               coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------
const                      -25.6420      5.815     -4.409      0.000     -37.040     -14.244
review_scores_rating         0.2592      0.059      4.378      0.000       0.143       0.375
number_of_reviews            0.0148      0.005      3.266      0.001       0.006       0.024
neighborhood_Broadway       -1.1647      0.385     -3.022      0.003      -1.920      -0.409
neighborhood_Fremont        -0.5164      0.461     -1.119      0.263      -1.421       0.388
neighborhood_Wallingford    -0.5754      0.452     -1.274      0.203      -1.461       0.310
room_type_Private room       0.3025      0.445      0.680      0.496      -0.569       1.174
room_type_Shared room       -1.2460      1.112     -1.120      0.263      -3.426       0.934
property_type_House         -0.3330      0.355     -0.938      0.348      -1.029       0.363
============================================================================================
"""

In [39]:
# Checking the data types of the reduced training dataset
X_train_reduced.dtypes

const                       float64
review_scores_rating          int64
number_of_reviews             int64
neighborhood_Broadway         int64
neighborhood_Fremont          int64
neighborhood_Wallingford      int64
room_type_Private room        int64
room_type_Shared room         int64
property_type_House           int64
dtype: object

In [41]:
# Dropping the variable with the highest p-value
X_train_reduced = X_train_reduced.drop(columns='room_type_Private room')
X_test_reduced = X_test_reduced.drop(columns='room_type_Private room')

# Refitting the logistic regression model without 'room_type_Private room'
log_reg_reduced_corrected = sm.Logit(y_train, X_train_reduced).fit()

# Showing the summary output of the further reduced logistic regression model
log_reg_reduced_corrected.summary()


Optimization terminated successfully.
         Current function value: 0.485542
         Iterations 8


<class 'statsmodels.iolib.summary.Summary'>
"""
                           Logit Regression Results                           
==============================================================================
Dep. Variable:      host_is_superhost   No. Observations:                  271
Model:                          Logit   Df Residuals:                      263
Method:                           MLE   Df Model:                            7
Date:                Tue, 31 Oct 2023   Pseudo R-squ.:                  0.1864
Time:                        12:20:22   Log-Likelihood:                -131.58
converged:                       True   LL-Null:                       -161.74
Covariance Type:            nonrobust   LLR p-value:                 1.309e-10
============================================================================================
                               coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------
const                      -26.0336      5.809     -4.482      0.000     -37.419     -14.648
review_scores_rating         0.2634      0.059      4.454      0.000       0.147       0.379
number_of_reviews            0.0150      0.005      3.329      0.001       0.006       0.024
neighborhood_Broadway       -1.1215      0.379     -2.961      0.003      -1.864      -0.379
neighborhood_Fremont        -0.5511      0.459     -1.200      0.230      -1.451       0.349
neighborhood_Wallingford    -0.5968      0.450     -1.326      0.185      -1.479       0.285
room_type_Shared room       -1.3207      1.105     -1.196      0.232      -3.486       0.844
property_type_House         -0.2436      0.329     -0.741      0.459      -0.888       0.401
============================================================================================
"""

In [42]:
# Calculating the predictive probabilities for the test dataset using the further reduced model
phat_test_further_reduced = log_reg_reduced_corrected.predict(X_test_reduced)

# Calculating the test log-likelihood function value for the further reduced model
test_llf_further_reduced = log_loss(y_test, phat_test_further_reduced)
test_llf_further_reduced


0.4976005749954986

### Comparison:

   * Test LLF of Previous Reduced Model: $0.5065$
   * Test LLF of Further Reduced Model: $0.4976$

Since the further reduced model has a lower (more negative) test log-likelihood function value than the previous reduced model, we will stop the backwards elimination algorithm here. The previous reduced model (before removing `room_type_Private room`) is the selected model.

This means that the model with the following variables is our final selected logistic regression model:

  *  `const`
  *  `review_scores_rating`
  *  `number_of_reviews`
  *  `neighborhood_Broadway`
  *  `neighborhood_Fremont`
  *  `neighborhood_Wallingford`
  *  `room_type_Shared room`
  *  `property_type_House`

This final model has a test log-likelihood function value of $0.5065$, indicating its performance on the test dataset.​​

## 4. Final Model

Write out your final logistic regression model in equation form using the three different representations that we discussed in class.
1. The one that predicts the **probability** the host is a superhost on the left hand side.
2. The one that predicts the **odds** the host is a superhost on the left hand side.
3. The one that predicts the **log-odds** the host is a superhost on the left hand side.

Make sure that you use the appropriate notation that we discussed in class.

The final model includes the following variables and their respective coefficients:
- Intercept ($ \beta_0 $): -25.6420
- review_scores_rating ($ \beta_1 $): 0.2592
- number_of_reviews ($ \beta_2 $): 0.0148
- neighborhood_Broadway ($ \beta_3 $): -1.1647
- neighborhood_Fremont ($ \beta_4 $): -0.5164
- neighborhood_Wallingford ($ \beta_5 $): -0.5754
- room_type_Private room ($ \beta_6 $): 0.3025
- room_type_Shared room ($ \beta_7 $): -1.2460
- property_type_House ($ \beta_8 $): -0.3330

### 1. Probability Representation:
The probability that a host is a superhost is given by:

$$
P(Y = 1) = \frac{1}{1 + \exp(-(\beta_0 + \beta_1 \cdot \text{{review\_scores\_rating}} + \beta_2 \cdot \text{{number\_of\_reviews}} + \beta_3 \cdot \text{{neighborhood\_Broadway}} + \beta_4 \cdot \text{{neighborhood\_Fremont}} + \beta_5 \cdot \text{{neighborhood\_Wallingford}} + \beta_6 \cdot \text{{room\_type\_Private room}} + \beta_7 \cdot \text{{room\_type\_Shared room}} + \beta_8 \cdot \text{{property\_type\_House}}))}
$$

### 2. Odds Representation:
The odds that a host is a superhost is given by:

$$
\frac{P(Y = 1)}{1 - P(Y = 1)} = \exp(\beta_0 + \beta_1 \cdot \text{{review\_scores\_rating}} + \beta_2 \cdot \text{{number\_of\_reviews}} + \beta_3 \cdot \text{{neighborhood\_Broadway}} + \beta_4 \cdot \text{{neighborhood\_Fremont}} + \beta_5 \cdot \text{{neighborhood\_Wallingford}} + \beta_6 \cdot \text{{room\_type\_Private room}} + \beta_7 \cdot \text{{room\_type\_Shared room}} + \beta_8 \cdot \text{{property\_type\_House}})
$$

### 3. Log-Odds Representation:
The log-odds that a host is a superhost is given by:

$$
\log\left(\frac{P(Y = 1)}{1 - P(Y = 1)}\right) = \beta_0 + \beta_1 \cdot \text{{review\_scores\_rating}} + \beta_2 \cdot \text{{number\_of\_reviews}} + \beta_3 \cdot \text{{neighborhood\_Broadway}} + \beta_4 \cdot \text{{neighborhood\_Fremont}} + \beta_5 \cdot \text{{neighborhood\_Wallingford}} + \beta_6 \cdot \text{{room\_type\_Private room}} + \beta_7 \cdot \text{{room\_type\_Shared room}} + \beta_8 \cdot \text{{property\_type\_House}}
$$

In these equations:
- $ P(Y = 1) $ represents the probability that the host is a superhost.
- The variables (e.g., review_scores_rating, number_of_reviews, etc.) represent the corresponding values for a particular Airbnb listing.
- The coefficients (e.g., $ \beta_0, \beta_1, \ldots $) represent the estimated coefficients from the logistic regression model.

These representations help in understanding how changes in the explanatory variables are associated with changes in the probability, odds, and log-odds of being a superhost.



Sophia Bustos, Josh Agrest, and myself all worked equally and collabrated on this lab.